File: baekjoon/최고빈출 DFS, BFS 기본문제/1260_DFS와 BFS.py
from collections import deque

# n: 정점의 개수, m: 간선의 개수, v: 시작 정점
n, m, v = map(int, input().split())
graph = [[] for _ in range(n+1)]
visited = [False for _ in range(n+1)] #각 노드가 방문된 정보를 표현

for i in range(m): #중요한 점! sort로 정렬해주고 stack이나 queue에 넣어야 한다.
    start, end = map(int, input().split())
    graph[start].append(end)
    graph[end].append(start)
    graph[start].sort()
    graph[end].sort()

def DFS(graph, start, visited):
    visited[start] = True #현재 노드를 방문 처리
    print(start, end=' ')

    for i in graph[start]: #현재 노드와 연결된 다른 노드를 재귀적으로 방문
        if not visited[i]:
            DFS(graph, i, visited)

def BFS(graph, start, visited):
    queue = deque([start]) #큐 구현을 위해 deque라이브러리 사용
    visited[start] = True #현재 노드를 방문 처리
    while queue: #큐가 빌 때까지 반복
        vertex = queue.popleft() #큐에서 하나의 원소를 뽑아 출력하기
        print(vertex, end=" ")
        for i in graph[vertex]: #아직 방문하지 않은 인접한 원소들을 큐에 삽입
            if not visited[i]:
                queue.append(i)
                visited[i] = True

DFS(graph, v, visited)
visited = [False for _ in range(n+1)] #각 노드가 방문된 정보를 표현
print()
BFS(graph, v, visited)

File: baekjoon/7.그래프의탐색(DFS,BFS)/1260_DFS와BFS.js
const fs = require('fs');
let input = fs.readFileSync("../test.txt").toString().trim().split('\n');
let NMV = input.shift().split(' ').map(function (n) {
    return parseInt(n, 10);
})
const N = NMV.shift();
const M = NMV.shift();
const V = NMV.shift();

let visited = [];
let edge = [];
let answer = [];

function dfs(start) {
    visited[start] = true;
    answer.push(start);

    for (let i = 0; i < edge[start].length; i++) {
        let next = edge[start][i];
        if (next && !visited[next]) {
            dfs(next);
        }
    }
}

function bfs(start) {
    let queue = [];
    visited[start] = true;
    queue.push(start);
    answer.push(start);

    while (queue.length !== 0) {
        let current = queue.shift();

        for (let i = 0; i < edge[current].length; i++) {
            let next = edge[current][i];
            if (next && !visited[next]) {
                queue.push(next);
                answer.push(next);
                visited[next] = true;
            }
        }
    }
}

function solution(N, M, V, input) {
    visited = new Array(N + 1).fill(false);
    for (let i = 1; i < N + 1; i++) {
        edge[i] = [];
    }
    for (let i = 0; i < M; i++) {
        let [from, to] = input[i].split(' ').map(function (n) {
            return parseInt(n, 10);
        });
        edge[from].push(to);
        edge[to].push(from);
    };

    edge.forEach((el) => {
        el.sort((a, b) => a - b);
    });

    dfs(V);
    console.log(answer.join(" "));

    visited = new Array(N + 1).fill(false);
    answer = [];

    bfs(V);
    console.log(answer.join(" "));
}

solution(N, M, V, input);